### PR TITLE
Handle deprecation of numpy `_validate_lengths`

### LINKED ISSUE #3551: _validate_lengths replaced in numpy/master leading to ImportError exception
## Description

As seen in https://travis-ci.org/astropy/astropy/jobs/455917279 , `from skimage.measure import block_reduce` caused `ImportError` when used with numpy dev.

```python
    from skimage.measure import block_reduce
  File ".../skimage/measure/__init__.py", line 3, in <module>
    from ._marching_cubes_classic import (marching_cubes_classic,
  File ".../skimage/measure/_marching_cubes_classic.py", line 3, in <module>
    from .._shared.utils import warn
  File ".../skimage/_shared/utils.py", line 10, in <module>
    from ..util import img_as_float
  File ".../skimage/util/__init__.py", line 8, in <module>
    from .arraycrop import crop
  File ".../skimage/util/arraycrop.py", line 8, in <module>
    from numpy.lib.arraypad import _validate_lengths
ImportError: cannot import name '_validate_lengths'
```
https://github.com/scikit-image/scikit-image/blob/29a22e17c8b4ccf26383e3007ab014eb18b81f12/skimage/util/arraycrop.py#L7

Caused by numpy/numpy#11966

## Way to reproduce
[If reporting a bug, please include the following important information:]
- [x] Code example
- [ ] Relevant images (if any)
- [x] Operating system and version -- Linux-4.14.12-041412-generic-x86_64-with-debian-jessie-sid
- [x] Python version -- 3.6.6
- [x] scikit-image version (run `skimage.__version__`) -- 0.14.0

All the info is in https://travis-ci.org/astropy/astropy/jobs/455917279 .

### PR DESCRIPTION
**Pull request by sciunto**
## Description

Fixes #3551 

I had to make a choice here. I preferred to keep using numpy. I removed the commented code (IMHO, not such a good practice to do that, we can browse numpy's history at any time)

A TODO task is added to remove the conditional function call.

This PR needs to be backported.

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](https://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/util/arraycrop.py [hunk 1]
```diff
@@ -4,138 +4,18 @@
...
-from numpy.lib.arraypad import _validate_lengths
 
+# After numpy 1.15, a new backward compatible function have been implemented.
+# See https://github.com/numpy/numpy/pull/11966
+old_numpy = False
+try:
+    from numpy.lib.arraypad import _as_pairs
+except ImportError:
+    from numpy.lib.arraypad import _validate_lengths
+    old_numpy = True
```
Comment (jni) [inline skimage/util/arraycrop.py]: @sciunto, I prefer to have the NumPy version check explicit here:

```
from distutils import LooseVersion as Version
old_numpy = Version(np.__version__) < Version('1.16')
if old_numpy:
    from numpy.lib.arraypad import _validate_lengths
else:
    from numpy.lib.arraypad import _as_pairs
```
This makes it clearer (though code) whether 1.15 is inclusive or exclusive.

Comment (sciunto) [reply skimage/util/arraycrop.py]: I made this change :)

**Comment by sciunto** [comment]
I agree with your observation. There are pro and cons. numpy does the job to maintain this code vs they may change again the behavior. I do not have strong feelings on this. my position was to fix it asap first.

File: skimage/util/arraycrop.py (hunk 1 above)
Comment (sciunto) [reply skimage/util/arraycrop.py]: it's distutils.version import... ;)

Comment (jni) [reply skimage/util/arraycrop.py]: Oops, sorry! =\

**Comment by jni** [comment]
> my position was to fix it asap first.

A very good position! =)

**Comment by soupault** [comment]
Thanks @sciunto !

**Comment by jriddy** [comment]
Any info on when this will get released on pypi? Numpy 1.16 just came out and broke this in our CI jobs. We can do a custom build of scikit-image internally for the time being but I prefer to keep with the mainstream of packages when possible

**Comment by jni** [comment]
@jriddy sorry about that! We are aiming to release 0.14.2 and 0.15 (Python 3.5+ only) in the next couple of weeks...

**Comment by jriddy** [comment]
@jni No reason to be sorry, we appreciate the work you guys do. 

Is [the backport branch for 0.14.x](https://github.com/scikit-image/scikit-image/tree/v0.14.x) currently _stable_ for some value of that word? I can easily clone this and cut an internal 0.14.2-dev1 release that can hold us over until you guys release to PyPI. I just want to know if your dev cycle includes a big QA blitz at the end before a release, or if you generally expect things to work well in your backport branch.

**Comment by jni** [comment]
Oh! Both v0.14.x and master should work great! We try not to merge PRs to either that will break things or require follow up work. (Other than maybe code style or minor speed issues, and even then only rarely.) The last minute pre-release push is just fiddling with release notes and version numbers.

**Comment by jriddy** [comment]
Hmmm, is there any guide here for building wheels for this project?

Locally my wheel is working, but inside my dockers I'm getting `E   ImportError: /usr/local/lib/python3.6/site-packages/skimage/restoration/_unwrap_1d.cpython-36m-x86_64-linux-gnu.so: undefined symbol: PyFPE_jbuf` issues.

I'm building in a virtualenv on Ubuntu 18.04 with python 3.6.7.

**Comment by stefanv** [comment]
On Mon, 14 Jan 2019 08:33:28 -0800, Josh Reed wrote:
> Hmmm, is there any guide here for building wheels for this project?
> 
> Locally my wheel is working, but inside my dockers I'm getting `E   ImportError: /usr/local/lib/python3.6/site-packages/skimage/restoration/_unwrap_1d.cpython-36m-x86_64-linux-gnu.so: undefined symbol: PyFPE_jbuf` issues.
> 
> I'm building in a virtualenv on Ubuntu 18.04 with python 3.6.7.

The wheel building process is a bit complicated; that's what we have
this repo for: https://github.com/scikit-image/scikit-image-wheels

That said, I just noticed that our daily wheels have not been building
for a while, otherwise I could just have pointed you to those:

https://travis-ci.org/scikit-image/scikit-image-wheels/

@Matthew-Brett Do you know whether [this
error](https://travis-ci.org/scikit-image/scikit-image-wheels/jobs/479187321#L4177)
is currently present for all multibuilds?

**Comment by jriddy** [comment]
It's okay, my original idea was to just mod the version and build a `scikit-image-0.14.1.post1` wheel and host it on our internal pypi so it would preempt 0.14.1 but get superseded automatically once you guys release 0.14.2 or 0.15.0, because that would have been the least amount of intervention in the context of my org's CI setup, but as it got more involved, it's just turned out to be simpler to pin numpy to <1.16 in a few of our repos and wait until you guys release.

Thanks for your responses, folks! I really appreciate it.

**Comment by pllim** [comment]
Hello. I am confused. I thought this is fixed in 0.14.2 but this PR is milestoned to 0.14.3.

**Comment by sciunto** [comment]
Because I renamed the milestone as several issues were still open after 0.14.2. I shouldn't have done that...  Sorry.

**Pull request merged**